app/Controllers/PerfilController.cs

```diff
@@ -0,0 +1,146 @@
+            this.mapper = mapper;
+            this.permissaoService = permissaoService;
+        }
+
+        [Authorize]
+        [HttpPost()]
+        public IActionResult CriarPerfil([FromBody] PerfilDTO perfilDTO)
+        {
+            authService.Require(Usuario, Permissao.PerfilCadastrar);
+
+            var perfil = mapper.Map<Perfil>(perfilDTO);
+
+            try{
+                Perfil novoPerfil = perfilService.CriarPerfil(perfil, perfilDTO.Permissoes);
+                return Ok(mapper.Map<PerfilModel>(novoPerfil));
+            }
+            catch(DbUpdateException)
+            {
+                return UnprocessableEntity("Este Perfil já existe");
+            }
+            catch(Exception)
+            {
+                return StatusCode(500, "Houve um erro interno no servidor.");
+            }
+        }
+
+        [Authorize]
+        [HttpPut("{id}")]
+        public async Task<IActionResult> EditarPerfil(Guid id, [FromBody] PerfilDTO perfilDTO)
+        {
+            authService.Require(Usuario, Permissao.PerfilEditar);
+
+            var perfil = mapper.Map<Perfil>(perfilDTO);
+            perfil.Id = id;
+
+            try{
+                var novoPerfil = await perfilService.EditarPerfil(perfil, perfilDTO.Permissoes);
+                return Ok(mapper.Map<PerfilModel>(novoPerfil));
+            }
+            catch(KeyNotFoundException)
+            {
+                return NotFound("Perfil não encontrado");
+            }
+            catch(DbUpdateException)
+            {
+                return UnprocessableEntity("Este Perfil já existe");
+            }
+            catch(Exception ex)
+            {
+                return StatusCode(500, $"Houve um erro interno no servidor. {ex.Message}");
+            }
+        }
+
+        [Authorize]
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> ExcluirPerfil(Guid id)
+        {
+            authService.Require(Usuario, Permissao.PerfilRemover);
+
+            try{
+                await perfilService.ExcluirPerfil(id);
+                return Ok("Perfil excluido");
+            }
+            catch(KeyNotFoundException ex){
+                return NotFound(ex.Message);
+            }
+            catch(InvalidOperationException e)
+            {
+                return StatusCode(400, e.Message);
+            }
+            catch(Exception)
+            {
+                return StatusCode(500, "Houve um erro interno no servidor.");
+            }            
+        }
+
+        [HttpGet]
+        [Authorize]
+        public async Task<IActionResult> ListarPerfis(int pageIndex, int pageSize, string? nome = null)
+        {
+            authService.Require(Usuario, Permissao.PerfilVisualizar);
+
+            try
+            {
+                var pagina = await perfilService.ListarPerfisAsync(pageIndex, pageSize, nome);
+
+                List<PerfilModel> paginaRetorno = pagina.Select(p => mapper.Map<PerfilModel>(p)).ToList();
```

Comment:
```suggestion
                List<PerfilModel> paginaRetorno = mapper.Map<List<PerfilModel>, List<TipoDestino>>(pagina);
```

Confira a documentação: https://docs.automapper.org/en/stable/Lists-and-arrays.html